hashes/src/hkdf/mod.rs

```diff
@@ -106,6 +106,26 @@ where
     }
 }
 
+impl<T: GeneralHash> fmt::Debug for Hkdf<T> {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+        use crate::{sha256t, sha256t_tag};
+
+        struct Fingerprint([u8; 8]); // Print 16 hex characters as a fingerprint.
+
+        impl fmt::Debug for Fingerprint {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { crate::debug_hex(&self.0, f) }
+        }
```

Comment:
Seeing this and similar code in `Witness` makes me think it might be worth it to have `WrapDebug<F: Fn(&mut fmt::Formatter) -> fmt::Result>(F);` in `internals` and use that with an ad-hoc closure.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/issues/4074